src/middlewared/middlewared/api/base/types/user.py

```diff
@@ -39,7 +39,7 @@ def validate_username(
     if valid_start_chars is not None:
         assert val[0] in valid_start_chars, 'Username must start with a letter or an underscore'
 
-    assert '$' not in val or val[-1] == '$', 'Username must end with a dollar sign character'
+    assert '$' not in val or val[-1] == '$', 'Username must not end with a dollar sign character'
```

Comment:
Not sure this is correct.  What rule are you trying to enforce?

Reply:
Oh. Our validation is weird I looked at that wrong. I thought we were trying to avoid having samba machine account names.